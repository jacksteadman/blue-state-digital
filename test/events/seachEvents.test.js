import nock from 'nock'
import path from 'path'
import fs from 'fs'
import BSD from './../../src'

require('dotenv').config()

beforeAll(() => {
  nock(`https://${process.env.BSD_DEV}`)
    .get('/page/api/event/search_events')
    .query(true)
    .reply(200, (uri, requestBody, cb) => {
      fs.readFile(path.resolve(__dirname, '../fixtures/events/searchEvents.json'), cb)
    })

  nock(`https://${process.env.BSD_DEV}`)
    .get('/page/api/event/search_events')
    .query(true)
    .reply(200, (uri, requestBody, cb) => {
      fs.readFile(path.resolve(__dirname, '../fixtures/events/searchEvents.json'), cb)
    })
})

afterAll(() => {
  nock.cleanAll()
})

const blueStateDigital = new BSD({
  baseUrl: process.env.BSD_DEV,
  apiID: process.env.BSD_API_ID,
  apiSecret: process.env.BSD_API_SECRET,
  apiVer: 2,
})

describe('#searchAllEvents()', () => {
  it('should have the property "searchEvents"', () => {
    expect(blueStateDigital).toHaveProperty('searchEvents')
  })

  it('should return all past and future events', async () => {
    const events = await blueStateDigital.searchEvents()
    expect(events).toBeDefined()
    expect(events).toMatchSnapshot()
  })

  it('should return one event by the event id', async () => {
    const event = await blueStateDigital.searchEvents({ event_id: '457' })
    expect(event).toBeDefined()
    expect(event).toHaveLength(1)
    expect(event).toMatchSnapshot()
  })
})
