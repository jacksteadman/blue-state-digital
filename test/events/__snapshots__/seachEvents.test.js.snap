// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#searchAllEvents() should return all past and future events 1`] = `
Array [
  Object {
    "_children": null,
    "_event_type": null,
    "_lastShiftFound": -1,
    "_shiftcount": -1,
    "_type": null,
    "all_shifts_full": "0",
    "attendee_count": "2",
    "attendee_require_phone": "0",
    "attendee_visibility": "2",
    "attendee_volunteer_message": "",
    "attendee_volunteer_show": "0",
    "capacity": "0",
    "chapter_id": "1",
    "chapter_name": null,
    "closed_msg": null,
    "contact_phone": "(555)555-5555",
    "contribution_page_id": null,
    "country_name": null,
    "county_name": null,
    "create_app": null,
    "create_dt": "2017-12-12 01:10:52",
    "create_user": null,
    "creator_cons_id": "212920",
    "creator_name": null,
    "delete_hooks": Array [
      "clearShifts",
      "hookPickNewParentOnDelete",
    ],
    "description": "Example Description",
    "dictionary": Object {},
    "duration": "120",
    "event_id": "457",
    "event_id_obfuscated": "wrl4",
    "event_type_id": "3",
    "event_type_name": "Canvass",
    "flag_approval": "0",
    "group_name": null,
    "guest_total": null,
    "hook_date_override": null,
    "hooksEnabled": true,
    "host_addr_addr1": "Another dimension,",
    "host_addr_addr2": "",
    "host_addr_addressee": "Rick Sanchez",
    "host_addr_city": "Made Up Place",
    "host_addr_country": "US",
    "host_addr_state_cd": "WA",
    "host_addr_zip": "78748",
    "host_name": null,
    "host_receive_rsvp_emails": "1",
    "is_official": "0",
    "is_searchable": "1",
    "latitude": "46.263700",
    "longitude": "-200.714798",
    "map_marker_storage_key": null,
    "modified_app": null,
    "modified_dt": "2017-12-12 06:33:17",
    "modified_user": null,
    "multiple_day_iteration": 0,
    "name": "Example Event",
    "note": null,
    "outreach_page_id": null,
    "override_url": null,
    "parent_event_id": "0",
    "pledge_max": "0",
    "pledge_min": "0",
    "pledge_override_type": "0",
    "pledge_require": "0",
    "pledge_show": "0",
    "pledge_source": null,
    "pledge_subsource": null,
    "pledge_suggest": "0",
    "post_save_hooks": Array [
      "expireShiftCache",
      "hook_fill_obfuscated_id",
      "hookReorderMultidayEvents",
    ],
    "public_phone": "0",
    "radius_unit": null,
    "rsvp_allow": "-1",
    "rsvp_disallow_account": "-1",
    "rsvp_email_message": null,
    "rsvp_email_message_html": null,
    "rsvp_email_reminder_hours": "24",
    "rsvp_reason": null,
    "rsvp_redirect_url": "",
    "rsvp_reminder_email_sent": "0",
    "rsvp_require_signup": "-1",
    "rsvp_use_default_email_message": "1",
    "rsvp_use_reminder_email": "1",
    "save_hooks": Array [
      "on_save",
      "geocode",
    ],
    "start_day": "2017-12-15",
    "start_dt": "2017-12-15 22:15:00",
    "start_time": "22:15:00",
    "start_tz": "US/Central",
    "status": "1",
    "tableFields": Array [
      "event_id_obfuscated",
      "event_type_id",
      "creator_cons_id",
      "contribution_page_id",
      "outreach_page_id",
      "name",
      "description",
      "creator_name",
      "start_day",
      "start_time",
      "start_dt",
      "start_tz",
      "duration",
      "parent_event_id",
      "venue_name",
      "venue_addr1",
      "venue_addr2",
      "venue_zip",
      "venue_city",
      "venue_state_cd",
      "venue_country",
      "venue_directions",
      "latitude",
      "longitude",
      "host_addr_addressee",
      "host_addr_addr1",
      "host_addr_addr2",
      "host_addr_zip",
      "host_addr_city",
      "host_addr_state_cd",
      "host_addr_country",
      "host_receive_rsvp_emails",
      "contact_phone",
      "public_phone",
      "capacity",
      "all_shifts_full",
      "closed_msg",
      "attendee_visibility",
      "attendee_require_phone",
      "attendee_volunteer_show",
      "attendee_volunteer_message",
      "is_official",
      "pledge_override_type",
      "pledge_show",
      "pledge_source",
      "pledge_subsource",
      "pledge_require",
      "pledge_min",
      "pledge_max",
      "pledge_suggest",
      "rsvp_use_default_email_message",
      "rsvp_email_message",
      "rsvp_email_message_html",
      "rsvp_use_reminder_email",
      "rsvp_reminder_email_sent",
      "rsvp_email_reminder_hours",
      "rsvp_allow",
      "rsvp_require_signup",
      "rsvp_disallow_account",
      "rsvp_reason",
      "rsvp_redirect_url",
      "is_searchable",
      "flag_approval",
      "chapter_id",
      "create_dt",
      "create_app",
      "create_user",
      "modified_dt",
      "modified_app",
      "modified_user",
      "status",
      "note",
    ],
    "tableName": "event",
    "tablePrimary": "event_id",
    "venue_addr1": "1203 Chicon St",
    "venue_addr2": "",
    "venue_city": "Made Up Place",
    "venue_country": "US",
    "venue_directions": "",
    "venue_name": "Rio Rita",
    "venue_state_cd": "WA",
    "venue_zip": "78701",
    "zip_radius": null,
  },
]
`;

exports[`#searchAllEvents() should return one event by the event id 1`] = `
Array [
  Object {
    "_children": null,
    "_event_type": null,
    "_lastShiftFound": -1,
    "_shiftcount": -1,
    "_type": null,
    "all_shifts_full": "0",
    "attendee_count": "2",
    "attendee_require_phone": "0",
    "attendee_visibility": "2",
    "attendee_volunteer_message": "",
    "attendee_volunteer_show": "0",
    "capacity": "0",
    "chapter_id": "1",
    "chapter_name": null,
    "closed_msg": null,
    "contact_phone": "(555)555-5555",
    "contribution_page_id": null,
    "country_name": null,
    "county_name": null,
    "create_app": null,
    "create_dt": "2017-12-12 01:10:52",
    "create_user": null,
    "creator_cons_id": "212920",
    "creator_name": null,
    "delete_hooks": Array [
      "clearShifts",
      "hookPickNewParentOnDelete",
    ],
    "description": "Example Description",
    "dictionary": Object {},
    "duration": "120",
    "event_id": "457",
    "event_id_obfuscated": "wrl4",
    "event_type_id": "3",
    "event_type_name": "Canvass",
    "flag_approval": "0",
    "group_name": null,
    "guest_total": null,
    "hook_date_override": null,
    "hooksEnabled": true,
    "host_addr_addr1": "Another dimension,",
    "host_addr_addr2": "",
    "host_addr_addressee": "Rick Sanchez",
    "host_addr_city": "Made Up Place",
    "host_addr_country": "US",
    "host_addr_state_cd": "WA",
    "host_addr_zip": "78748",
    "host_name": null,
    "host_receive_rsvp_emails": "1",
    "is_official": "0",
    "is_searchable": "1",
    "latitude": "46.263700",
    "longitude": "-200.714798",
    "map_marker_storage_key": null,
    "modified_app": null,
    "modified_dt": "2017-12-12 06:33:17",
    "modified_user": null,
    "multiple_day_iteration": 0,
    "name": "Example Event",
    "note": null,
    "outreach_page_id": null,
    "override_url": null,
    "parent_event_id": "0",
    "pledge_max": "0",
    "pledge_min": "0",
    "pledge_override_type": "0",
    "pledge_require": "0",
    "pledge_show": "0",
    "pledge_source": null,
    "pledge_subsource": null,
    "pledge_suggest": "0",
    "post_save_hooks": Array [
      "expireShiftCache",
      "hook_fill_obfuscated_id",
      "hookReorderMultidayEvents",
    ],
    "public_phone": "0",
    "radius_unit": null,
    "rsvp_allow": "-1",
    "rsvp_disallow_account": "-1",
    "rsvp_email_message": null,
    "rsvp_email_message_html": null,
    "rsvp_email_reminder_hours": "24",
    "rsvp_reason": null,
    "rsvp_redirect_url": "",
    "rsvp_reminder_email_sent": "0",
    "rsvp_require_signup": "-1",
    "rsvp_use_default_email_message": "1",
    "rsvp_use_reminder_email": "1",
    "save_hooks": Array [
      "on_save",
      "geocode",
    ],
    "start_day": "2017-12-15",
    "start_dt": "2017-12-15 22:15:00",
    "start_time": "22:15:00",
    "start_tz": "US/Central",
    "status": "1",
    "tableFields": Array [
      "event_id_obfuscated",
      "event_type_id",
      "creator_cons_id",
      "contribution_page_id",
      "outreach_page_id",
      "name",
      "description",
      "creator_name",
      "start_day",
      "start_time",
      "start_dt",
      "start_tz",
      "duration",
      "parent_event_id",
      "venue_name",
      "venue_addr1",
      "venue_addr2",
      "venue_zip",
      "venue_city",
      "venue_state_cd",
      "venue_country",
      "venue_directions",
      "latitude",
      "longitude",
      "host_addr_addressee",
      "host_addr_addr1",
      "host_addr_addr2",
      "host_addr_zip",
      "host_addr_city",
      "host_addr_state_cd",
      "host_addr_country",
      "host_receive_rsvp_emails",
      "contact_phone",
      "public_phone",
      "capacity",
      "all_shifts_full",
      "closed_msg",
      "attendee_visibility",
      "attendee_require_phone",
      "attendee_volunteer_show",
      "attendee_volunteer_message",
      "is_official",
      "pledge_override_type",
      "pledge_show",
      "pledge_source",
      "pledge_subsource",
      "pledge_require",
      "pledge_min",
      "pledge_max",
      "pledge_suggest",
      "rsvp_use_default_email_message",
      "rsvp_email_message",
      "rsvp_email_message_html",
      "rsvp_use_reminder_email",
      "rsvp_reminder_email_sent",
      "rsvp_email_reminder_hours",
      "rsvp_allow",
      "rsvp_require_signup",
      "rsvp_disallow_account",
      "rsvp_reason",
      "rsvp_redirect_url",
      "is_searchable",
      "flag_approval",
      "chapter_id",
      "create_dt",
      "create_app",
      "create_user",
      "modified_dt",
      "modified_app",
      "modified_user",
      "status",
      "note",
    ],
    "tableName": "event",
    "tablePrimary": "event_id",
    "venue_addr1": "1203 Chicon St",
    "venue_addr2": "",
    "venue_city": "Made Up Place",
    "venue_country": "US",
    "venue_directions": "",
    "venue_name": "Rio Rita",
    "venue_state_cd": "WA",
    "venue_zip": "78701",
    "zip_radius": null,
  },
]
`;
